### PR TITLE
Long running tasks

### PR DESCRIPTION
**Pull request by honza**
See the README for more info.

To test, start the server, and open the app. Then, in another browser window or with curl, you can hit the http://localhost:8080/api/long endpoint. You should see messages printed out from the server process, and in the developer console in the browser.

Rebased on top of #11

### CODE REVIEW

**Comment by knowncitizen** [comment]
I don't think we want vendor binaries?

**Comment by honza** [comment]
* Rebased again on top of #11

**Comment by honza** [comment]
* Rebased on top of `master` now that #11 is merged

File: README.md [hunk 1]
```diff
@@ -70,6 +70,47 @@ Use "facet [command] --help" for more information about a command.
 
 [3]: https://github.com/rakyll/statik
 
+## Long-running tasks
+
+If you would like to trigger a long-running task from the frontend, we've got
```
Comment (knowncitizen) [inline README.md]: Neat.

**Comment by honza** [comment]
> Where does log go? Is it configurable?

By default, it's just printed to stdout from the server process. Yes, it can be configured --- what did you have in mind?

**Comment by knowncitizen** [comment]
Was just curious. Thanks for the insight.

**Comment by honza** [comment]
* Rebase on master

File: pkg/server/routes.go [hunk 2]
```diff
@@ -34,11 +35,45 @@ type ApiResponse struct {
...
+	}
+
+	return values[ns]
+}
+
+func (ns NotificationStatus) MarshalJSON() ([]byte, error) {
+	return json.Marshal(ns.String())
+}
+
+type Notification struct {
```
Comment (jtomasek) [inline pkg/server/routes.go]: At some point, we'll need to figure out standardized notification format, probably aligned to day2 (I expect 'type' to identify the message and 'payload' for additional data)

File: src/api/websocket.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// TODO: This is just a dumb client to test the websocket connection
+// TODO: This needs to be connected to the redux store
```
Comment (jtomasek) [inline src/api/websocket.ts]: We can experiment with putting this into a React Component which would handle websocket errors, reconnecting and dispatching the actions based on notification type

**Pull request merged**